Automattic/wp-calypso
# Nav Unification: Remove unused logic

#### Changes proposed in this Pull Request

Removes the Nav Unification logic that is no longer needed:
- The announcement modal, since the feature was released a few months ago.
- The `nav-unification/switcher` feature flag, since the quick switcher has been stable in all environments for a few weeks.

#### Testing instructions

- Make sure that no modal is displayed, even when the `nav-unification-modal-<USER_ID>` preference is unset <img width="847" alt="Screen Shot 2021-07-23 at 15 16 11" src="https://user-images.githubusercontent.com/1233880/126787355-7b0cb342-cca8-4b2a-b49c-35281f063124.png">
- Make sure the "Screen Options" tab is displayed in pages with an equivalent WP Admin counterpart screen (e.g. Posts, Media, etc).
